src/site/content/en/blog/smoothness/index.md

```diff
@@ -0,0 +1,563 @@
+batched visual updates and produce a visually complete animation frame within
+the deadline, every time. Note that this goal is entirely distinct from other
+important browser tasks such as loading content from the network quickly or
+executing JavaScript tasks efficiently.
+
+At some point, it can become too difficult to complete all visual updates within
+the allotted deadline assigned by the display. When this happens, the browser
+**drops a frame**. Your screen doesn't go black, it just repeats itself. You see
+the same visual update for a bit longer—the same animation frame that was
+presented at the previous frame opportunity.
+
+This actually happens often! It is not necessarily even perceptible, especially
+for static or document-like content, which is common on the web platform in
+particular. Dropped frames only become apparent when there are important visual
+updates, such as animations, for which we need a steady stream of animation
+updates to show smooth motion.
+
+## What impacts animation frames?
+
+Web developers can greatly impact the ability of a browser to quickly and
+efficiently render and present visual updates!
+
+Some examples:
+
+- Using content that is too large or resource intensive to decode quickly on the
+  target device.
+- [Using too many
+  layers](https://developers.google.com/web/fundamentals/performance/rendering/stick-to-compositor-only-properties-and-manage-layer-count)
+  requiring too much GPU memory.
+- Defining overly complex CSS styles or web animations.
+- Using design anti-patterns that disable fast rendering optimizations.
+- Too much JS work on the main thread, leading to long tasks that block certain
+  visual updates.
+
+But how can you know when an animation frame has slipped its deadline and caused
+a dropped frame?
+
+You may be aware of one method:
+[`requestAnimationFrame()`](https://developer.mozilla.org/docs/Web/API/window/requestAnimationFrame)
+polling. `requestAnimationFrame()`, or "rAF", tells the browser that you wish to
+perform an animation and asks for an opportunity to do so before the next paint
+stage of the rendering pipeline. If your callback function isn't called at the
+time you expect it, that means a paint wasn't executed, and one or more frames
+were skipped. By polling and counting how often rAF is called, you can compute a
+sort of "frames per second" (fps) metric.
+
+```js
+let frameTimes = [];
+function pollFramesPerSecond(now) {
+  frameTimes = [...frameTimes.filter(t => t > now - 1000), now];
+  requestAnimationFrame(pollFramesPerSecond);
+  console.log('Frames per second:', frameTimes.length);
+}
+requestAnimationFrame(pollFramesPerSecond);
+```
+
+{% Aside 'warning' %}
+  The code above is an anti pattern and is strongly discouraged!
+{% endAside %}
+
+Using `requestAnimationFrame()` polling is not a good idea for several reasons:
+
+- Every interested party (or script) has to set up its own polling loop.
```

Comment:
```suggestion
- Every script has to set up its own polling loop.
```

Reply:
Again, this may seem finicky, but can this be simplified to just "script"? It's OK if not, because you may be referring to something else besides scripts, but if you are, maybe spell it out.